### PR TITLE
update taskflow to v3.1.0

### PR DESCRIPTION
**Pull request by fasiondog**
update taskflow to v3.1.0

### CODE REVIEW

File: packages/h/httplib/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            package:add("deps", "brotli")
+        end
+    end)
+
+    add_deps("cmake")
+    on_install(function (package)
+        local configs = {}
+        table.insert(configs, "-DHTTPLIB_REQUIRE_OPENSSL=" .. (package:config("ssl") and "ON" or "OFF"))
+        table.insert(configs, "-DHTTPLIB_REQUIRE_ZLIB=" .. (package:config("zlib") and "ON" or "OFF"))
+        table.insert(configs, "-DHTTPLIB_REQUIRE_BROTLI=" .. (package:config("brotli") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/h/httplib/xmake.lua:31]: pic 和 shared 配置加一下，参考：

https://github.com/xmake-io/xmake-repo/blob/3242cfd9eabbf970ca414fc2dd55bc92aab96d1f/packages/l/libfive/xmake.lua#L23-L26

File: packages/h/httplib/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        end
+        if package:config("zlib") then
+            package:add("deps", "zlib")
+        end
+        if package:config("brotli") then
+            package:add("deps", "brotli")
+        end
+    end)
+
+    add_deps("cmake")
```
Comment (waruqi) [inline packages/h/httplib/xmake.lua:26]: 请放到 on_load 前面

File: packages/h/httplib/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package("httplib")
```
Comment (yamashi) [inline packages/h/httplib/xmake.lua:1]: This already exists: https://github.com/xmake-io/xmake-repo/blob/master/packages/c/cpp-httplib/xmake.lua

**Review by yamashi — CHANGES_REQUESTED** [review]

File: packages/h/httplib/xmake.lua (hunk 3 above)
Comment (waruqi) [reply packages/h/httplib/xmake.lua:1]: yeah, this is a duplicate package, please use cpp-httplib directly. @fasiondog

**Comment by fasiondog** [comment]
应该是我没先同步

**Comment by fasiondog** [comment]
OK了

**Pull request merged**